### ISSUE TITLE
When converting from binary to decimal with to_i(2), the result will be 0 if it contains multiple underscores.

### ISSUE THREAD

**Issue by naitoh** [jruby/jruby#7749]
**Environment Information**
  
```
$ jruby -v
jruby 9.4.2.0 (3.1.0) 2023-03-08 90d2913fda Java HotSpot(TM) 64-Bit Server VM 25.121-b13 on 1.8.0_121-b13 +jit [x86_64-darwin]
```
```
$ uname -a
Darwin MacBook-Air.local 22.4.0 Darwin Kernel Version 22.4.0: Mon Mar  6 21:00:41 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T8103 arm64
```

**Expected Behavior**

```
# ruby -v
ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [aarch64-linux-gnu]
# uname -a
Linux 0ff438a0da33 5.15.49-linuxkit #1 SMP PREEMPT Tue Sep 13 07:51:32 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
# irb
> '0000_0010'.to_i(2)
=> 2
> '000000000010'.to_i(2)
=> 2
> '0000_0000_0010'.to_i(2)
=> 2
```

**Actual Behavior**

```
> '0000_0010'.to_i(2)
=> 2
```
👌 
```
> '000000000010'.to_i(2)
=> 2
```
👌 
```
> '0000_0000_0010'.to_i(2)
=> 0
```
🙅

**Comment by enebo** [comment]
@naitoh fascinating.  It does not work on 9.3 either.  Perhaps it has never worked.

**Comment by headius** [comment]
If the underscores come after the value it works ok:

```
$ jruby -e "puts '000000000010__'.to_i(2)"
2
```

It seems to bail out when encountering the second `_`.

**Comment by headius** [comment]
This affects all bases when the leading character is zero, due to a bug in `squeezeZeroes`:

```
[] jruby93 $ ruby31 -e "puts '0000_0000_0010'.to_i(2)" 
2
[] jruby93 $ ruby31 -e "puts '0000_0000_0010'.to_i(8)"
8
[] jruby93 $ ruby31 -e "puts '0000_0000_0010'.to_i(10)"
10
[] jruby93 $ ruby31 -e "puts '0000_0000_0010'.to_i(16)"
16
[] jruby93 $ jruby -e "puts '0000_0000_0010'.to_i(2)"
0
[] jruby93 $ jruby -e "puts '0000_0000_0010'.to_i(8)"
0
[] jruby93 $ jruby -e "puts '0000_0000_0010'.to_i(10)"
0
[] jruby93 $ jruby -e "puts '0000_0000_0010'.to_i(16)"
0
```

Fix coming.

**Comment by headius** [comment]
Will be fixed by #7934.
